index.js

```diff
@@ -34,6 +34,11 @@ function verifyTaskSets(gulp, taskSets, skipArrays) {
 }
 
 function runSequence(gulp) {
+	// enable gulp lazy loading
+	if(typeof gulp === 'function') {
```

Comment:
I'm not sure why the extra level of indirection.  Why not simply do:
```js
if(gulp === undefined) {
    gulp = require('gulp');
}
```

Then the lower change can simply be `module.exports = runSequence;`